content/oci/oci.go

```diff
@@ -375,6 +376,71 @@ func (s *Store) writeIndexFile() error {
 	return os.WriteFile(s.indexPath, indexJSON, 0666)
 }
 
+// Referrers lists the descriptors of image or artifact manifests directly
+// referencing the given manifest descriptor.
+//
+// fn is called on the referrer results. If artifactType is not empty, only
+// referrers of the same artifact type are fed to fn.
+//
+// Reference: https://github.com/opencontainers/distribution-spec/blob/v1.1.0-rc3/spec.md#listing-referrers
+func (s *Store) Referrers(ctx context.Context, desc ocispec.Descriptor, artifactType string, fn func(referrers []ocispec.Descriptor) error) error {
+	var results []ocispec.Descriptor
+	predecessors, err := s.Predecessors(ctx, desc)
+	if err != nil {
+		return err
+	}
+	for _, node := range predecessors {
+		var nodeArtifactType string
+		switch node.MediaType {
+		case ocispec.MediaTypeImageManifest:
+			fetched, err := content.FetchAll(ctx, s, node)
+			if err != nil {
+				return err
+			}
+			var manifest ocispec.Manifest
+			if err := json.Unmarshal(fetched, &manifest); err != nil {
+				return err
+			}
+			if manifest.Subject == nil || !content.Equal(*manifest.Subject, desc) {
+				continue
+			}
+			nodeArtifactType = manifest.MediaType
```

Comment:
why it's not
```suggestion
			nodeArtifactType = manifest.ArtifactType
```

Reply:
Also what if the artifact type is [denoted by the config descriptor's media type](https://oras.land/docs/concepts/artifact#prior-art)?

Reply:
> Also what if the artifact type is [denoted by the config descriptor's media type](https://oras.land/docs/concepts/artifact#prior-art)?

changed L406 to `node.ArtifactType = manifest.Config.MediaType`

Reply:
Changed L407 to 

```
			if manifest.ArtifactType != "" {
				node.ArtifactType = manifest.ArtifactType
			} else {
				node.ArtifactType = manifest.Config.MediaType
			}
			node.Annotations = manifest.Annotations
```